classes/Collection.js

```diff
@@ -111,17 +106,19 @@ class Collection {
     try {
       const commitMessage = `Rename collection from ${oldCollectionName} to ${newCollectionName}`
       // Rename collection in config
-      const config = new Config(this.accessToken, this.siteName)
-      const { content, sha } = await config.read()
-      const contentObject = yaml.safeLoad(base64.decode(content))
-
-      contentObject.collections[`${newCollectionName}`] = {
-        output: true 
+      const oldConfig = new CollectionConfig(this.accessToken, this.siteName, oldCollectionName)
+      const { content: oldConfigContent, sha: oldConfigSha } = await oldConfig.read()
+      const oldConfigContentObject = yaml.safeLoad(base64.decode(oldConfigContent))
+      await oldConfig.delete(oldConfigSha)
+
+      const newConfig = new CollectionConfig(this.accessToken, this.siteName, newCollectionName)
+      const newConfigContentObject = {
+        collections: {
+          [newCollectionName]: oldConfigContentObject.collections[`${oldCollectionName}`]
+        }
       }
-      delete contentObject.collections[`${oldCollectionName}`]
-      const newContent = base64.encode(yaml.safeDump(contentObject))
-
-      await config.update(newContent, sha)
+      const newConfigContent = base64.encode(yaml.safeDump(newConfigContentObject))
+      await newConfig.create(newConfigContent)
```

Comment:
As discussed, the git tree updates helps to rename the `collection.yml` config file as well, so it might make sense to update the config file's `collection` attribute value after we update the git tree. Alternatively we can explore other options

Reply:
resolved in fff5889